gapic-common/lib/gapic/rest/client_stub.rb

```diff
@@ -127,6 +129,10 @@ def make_put_request uri:, body: nil, params: {}, options: {}
       #   Currently only timeout and headers are supported.
       # @return [Faraday::Response]
       def make_http_request verb, uri:, body:, params:, options:
+        if @numeric_enums && (!params.key?("$alt") || params["$alt"] == "json")
+          params = params.merge({"$alt" => "json;enum-encoding=int"})
+        end
+        options = ::Gapic::CallOptions.new(**options.to_h) unless options.is_a? ::Gapic::CallOptions
```

Comment:
options should already be ::Gapic::CallOptions since it's a private method intended to only be called from service stubs.

Reply:
I understand it's intended to be a private interface. But unfortunately, it's not: ClientStub is a public, documented class. While `make_http_request` itself is marked `@private` and so isn't documented in YARD, other methods such as `make_get_request` are public, and it's actually that case that I'm concerned here with. Because `make_get_request` takes `options:` as a keyword argument and the default value is the empty hash. So currently, if you call `make_get_request` with the default arguments, it will pass an empty hash to `make_http_request` which will then raise NoMethodError because you can't call `.metadata` on a hash.

Reply:
ok, makes sense. let's change the docs here to indicate Hash is supported then.

Reply:
Done